prov/sm2/src/sm2_progress.c

```diff
@@ -144,7 +144,8 @@ static int sm2_alloc_xfer_entry_ctx(struct sm2_ep *ep,
 		return -FI_ENOMEM;
 	}
 
-	memcpy(&xfer_ctx->xfer_entry, xfer_entry, sizeof(*xfer_entry));
+	memcpy(&xfer_ctx->xfer_entry, xfer_entry,
+	       sizeof(struct sm2_xfer_hdr) + xfer_entry->hdr.size);
```

Comment:
Change the sizeof to get the size of the field to stay consistent with the rest of the code
`sizeof(xfer_entry->hdr)`